test-framework/junit5-component/src/main/java/io/quarkus/test/component/QuarkusComponentTestExtension.java

```diff
@@ -1203,6 +1247,25 @@ private static boolean isTypeArgumentInstanceHandle(java.lang.reflect.Type type)
         return false;
     }
 
+    private boolean injectionPointMatchesBean(java.lang.reflect.Type injectionPointType, AnnotatedElement annotatedElement,
+            List<DotName> allQualifiers, BeanResolver beanResolver, BeanInfo bean) {
+        Type requiredType;
+        Set<AnnotationInstance> requiredQualifiers = getQualifiers(annotatedElement, allQualifiers);
+        if (isListAllInjectionPoint(injectionPointType,
+                Arrays.stream(annotatedElement.getAnnotations())
+                        .filter(a -> allQualifiers.contains(DotName.createSimple(a.annotationType())))
+                        .toArray(Annotation[]::new),
+                annotatedElement)) {
+            requiredType = Types.jandexType(getFirstActualTypeArgument(injectionPointType));
+            adaptListAllQualifiers(requiredQualifiers);
+        } else if (Instance.class.isAssignableFrom(QuarkusComponentTestConfiguration.getRawType(injectionPointType))) {
```

Comment:
With this, any `Instance<Object>` injection point basically disables removal of unused beans - should we special case that?

Reply:
I don't think so. It works the same way in regular Quarkus apps...

Reply:
Hm, I didn't realize (or already forgot?) we do this in Quarkus as well, but you are correct.